### PR TITLE
AgentCore dynamic installation and uninstallation test: test plug-in, test application, pipeline

### PR DESCRIPTION
**Pull request by TangLeDaily**
【Fix issue】#1236

【Modified content】
1. Added some codes for automated integration testing of dynamic installation and uninstallation: test plug-ins, test applications, pipelines, etc.
2. Optimized the test agentcore test code structure

[Use case description] Some test cases, to be incorporated into subsequent code

[Self-test status] Local static check passed, local automated test passed

[Scope of influence] None

### CODE REVIEW

File: sermant-integration-tests/agentcore-test/agentcore-integration-test/src/test/java/com/huaweicloud/agentcore/test/request/utils/RequestUtils.java [hunk 1]
```diff
@@ -14,45 +14,41 @@
...
  */
-public class AgentcoreTest {
-    private static final Logger LOGGER = LoggerFactory.getLogger(AgentcoreTest.class);
-
-    @Test
-    public void testAgentcore() throws IOException {
-        testRequest("http://127.0.0.1:8915/testDynamicConfig");
-    }
+public class RequestUtils {
+    private static final Logger LOGGER = LoggerFactory.getLogger(DynamicConfigTest.class);
```
Comment (daizhenyu) [inline sermant-integration-tests/agentcore-test/agentcore-integration-test/src/test/java/com/huaweicloud/agentcore/test/request/utils/RequestUtils.java]: 这里为啥写DynamicConfigTest.class

File: sermant-integration-tests/agentcore-test/agentcore-test-application/src/main/java/com/huaweicloud/agentcore/test/application/controller/TestController.java [hunk 2]
```diff
@@ -60,6 +61,60 @@ public String testDynamicConfig() {
...
+     *
+     * @return 测试结果
+     */
+    public String testInstallPlugin() {
+        Map<String, Object> resultMap = new HashMap<>();
+        resultMap.put(DynamicResults.DYNAMIC_INSTALL_REPEAT_ENHANCE.name(),
+                DynamicResults.DYNAMIC_INSTALL_REPEAT_ENHANCE.getResult());
+        JSONObject jsonObject = new JSONObject(resultMap);
+        return jsonObject.toJSONString();
+    }
```
Comment (daizhenyu) [inline sermant-integration-tests/agentcore-test/agentcore-test-application/src/main/java/com/huaweicloud/agentcore/test/application/controller/TestController.java:78]: 这一块是接口方法吗?

File: sermant-integration-tests/agentcore-test/agentcore-integration-test/src/test/java/com/huaweicloud/agentcore/test/request/utils/RequestUtils.java (hunk 1 above)
Comment (TangLeDaily) [reply sermant-integration-tests/agentcore-test/agentcore-integration-test/src/test/java/com/huaweicloud/agentcore/test/request/utils/RequestUtils.java]: 已修改

File: sermant-integration-tests/agentcore-test/agentcore-test-application/src/main/java/com/huaweicloud/agentcore/test/application/controller/TestController.java (hunk 2 above)
Comment (TangLeDaily) [reply sermant-integration-tests/agentcore-test/agentcore-test-application/src/main/java/com/huaweicloud/agentcore/test/application/controller/TestController.java:78]: 这一块是请求处理逻辑并且返回response。具体http请求的路由控制在ControllerHandler和RouterPath

File: .github/actions/scenarios/agentcore/agentcore-common/action.yml [hunk 3]
```diff
@@ -1,12 +1,12 @@
 name: "Agent-core Common Test"
-description: "Auto test for spring common, include nacos dynamic config"
+description: "Auto test for agent-core common, include nacos dynamic config"
```
Comment (luanwenfei-venus) [inline .github/actions/scenarios/agentcore/agentcore-common/action.yml:2]: 为什么要单独强调include nacos dynamic config

File: .github/actions/scenarios/agentcore/dynamic/action.yml [hunk 4]
```diff
@@ -0,0 +1,92 @@
+name: "Agent-core Dynamic Test"
+description: "Auto test for agent-core dynamic"
```
Comment (luanwenfei-venus) [inline .github/actions/scenarios/agentcore/dynamic/action.yml]: name: "Agent-core Dynamic Install/Uninstall Test"
description: "Auto test for agent-core dynamic install/uninstall"

File: sermant-integration-tests/agentcore-test/agentcore-integration-test/src/test/java/com/huaweicloud/agentcore/test/request/utils/RequestUtils.java [hunk 5]
```diff
@@ -14,45 +14,41 @@
...
 
 /**
- * agentcore测试方法，采用http请求调用方式测试
+ * http请求工具类
  *
  * @author tangle
- * @since 2023-09-07
+ * @since 2023-09-26
  */
-public class AgentcoreTest {
```
Comment (luanwenfei-venus) [inline sermant-integration-tests/agentcore-test/agentcore-integration-test/src/test/java/com/huaweicloud/agentcore/test/request/utils/RequestUtils.java]: 这个工具类，缺少私有的构造函数

File: .github/actions/scenarios/agentcore/agentcore-common/action.yml (hunk 3 above)
Comment (TangLeDaily) [reply .github/actions/scenarios/agentcore/agentcore-common/action.yml:2]: 目前只包含了动态配置的测试，后续添加其他agent测试时候，可以加上去

File: .github/actions/scenarios/agentcore/dynamic/action.yml (hunk 4 above)
Comment (TangLeDaily) [reply .github/actions/scenarios/agentcore/dynamic/action.yml]: 已修改

File: sermant-integration-tests/agentcore-test/agentcore-integration-test/src/test/java/com/huaweicloud/agentcore/test/request/utils/RequestUtils.java (hunk 5 above)
Comment (TangLeDaily) [reply sermant-integration-tests/agentcore-test/agentcore-integration-test/src/test/java/com/huaweicloud/agentcore/test/request/utils/RequestUtils.java]: 已修改

Comment (TangLeDaily) [reply sermant-integration-tests/agentcore-test/agentcore-integration-test/src/test/java/com/huaweicloud/agentcore/test/request/utils/RequestUtils.java]: 该类及xxxTest测试类有小型重构，所在位置有所改变

**Review by luanwenfei-venus — APPROVED** [review]

**Review by daizhenyu — APPROVED** [review]

**Pull request merged**